### PR TITLE
state.apply with pillar data and state.show_highstate

### PR DESCRIPTION
**Pull request by mateiw**
* add `state.apply` call that takes pillar data
* add `state.show_highstate` call

### CODE REVIEW

File: src/main/java/com/suse/salt/netapi/calls/modules/State.java [hunk 1]
```diff
@@ -28,4 +29,18 @@ private State() { }
...
     }
 
+    public static LocalCall<Map<String, Object>> apply(String mod,
+            Map<String, Object> pillar) {
+        LinkedHashMap args = new LinkedHashMap();
+        args.put("mods", Collections.singletonList(mod));
+        args.put("pillar", pillar);
+        return new LocalCall("state.apply", Optional.empty(), Optional.of(args),
+                new TypeToken<Map<String, Object>>() { });
+    }
```
Comment (renner) [inline src/main/java/com/suse/salt/netapi/calls/modules/State.java]: This method currently supports only a single state module, is that right? Why not just adding the `pillar` parameter to the existing apply method so you could give in a list of state modules? Then if we want we can have another method (without the `pillar` parameter) that could just call the other one, if you know what I mean.

**Comment by renner** [comment]
I actually already created a PR with `state.show_highstate` (#144). We can close it and keep this one or you can also just add commits in #144, as you prefer.

File: src/main/java/com/suse/salt/netapi/calls/modules/State.java [hunk 2]
```diff
@@ -18,14 +18,27 @@
...
+
+    public static LocalCall<Map<String, Object>> apply(List<String> mods,
+            Optional<Map<String, Object>> pillar) {
+        LinkedHashMap args = new LinkedHashMap();
         args.put("mods", mods);
-        return new LocalCall<>("state.apply", Optional.empty(), Optional.of(args),
+        if (pillar.isPresent()) {
+            args.put("pillar", pillar.get());
+        }
+        return new LocalCall("state.apply", Optional.empty(), Optional.of(args),
```
Comment (renner) [inline src/main/java/com/suse/salt/netapi/calls/modules/State.java]: Please fix warnings throughout this code:

1. the `LinkedHashMap` should be created with type arguments: `<String, Object>`?
2. the returned `LocalCall` should be parameterized, diamond operator?

**Comment by renner** [comment]
Let's merge this one, @mateiw, @lucidd?

File: src/main/java/com/suse/salt/netapi/calls/modules/State.java [hunk 3]
```diff
@@ -18,14 +18,30 @@
...
     public static LocalCall<Map<String, Object>> apply(String... mods) {
-        return apply(Arrays.asList(mods));
+        return apply(Arrays.asList(mods), Optional.empty(), Optional.empty());
+    }
+
+    public static LocalCall<Map<String, Object>> apply(List<String> mods,
+            Optional<Map<String, Object>> pillar, Optional<Boolean> queue) {
+        Map<String, Object> kwargs = new LinkedHashMap<>();
+        kwargs.put("mods", mods);
+        if (pillar.isPresent()) {
```
Comment (lucidd) [inline src/main/java/com/suse/salt/netapi/calls/modules/State.java]: use `ifPresent` for this and the one below to avoid `get`

Comment (renner) [inline src/main/java/com/suse/salt/netapi/calls/modules/State.java]: ok, fixed with 5026298796a68dd3eb1d6b7731e25475bf4e67fe.

**Pull request merged**